metacat-functional-tests/src/functionalTest/groovy/com/netflix/metacat/MetacatSmokeSpec.groovy

```diff
@@ -514,6 +514,21 @@ class MetacatSmokeSpec extends Specification {
         parts.get(0).dataMetadata != null
         partkeys.size() == 2
 
+        when:
+        partitionApi.getPartitionCount(catalogName, databaseName, tableName)
+        then:
+        thrown(MetacatNotSupportedException)
```

Comment:
this should throw UnsupportedOperationException?

Reply:
No. We should throw MetacatNotSupportedException.